pandas/core/dtypes/common.py

```diff
@@ -1905,20 +1906,20 @@ def _get_dtype_type(arr_or_dtype):
     elif isinstance(arr_or_dtype, CategoricalDtype):
         return CategoricalDtypeType
     elif isinstance(arr_or_dtype, DatetimeTZDtype):
-        return DatetimeTZDtypeType
+        return Timestamp
     elif isinstance(arr_or_dtype, IntervalDtype):
-        return IntervalDtypeType
+        return Interval
     elif isinstance(arr_or_dtype, PeriodDtype):
-        return PeriodDtypeType
+        return Period
     elif isinstance(arr_or_dtype, string_types):
         if is_categorical_dtype(arr_or_dtype):
```

Comment:
what was rationale for not making this object for Categorical here? otherwise this is the odd man out

Reply:
I don't recall the previous discussion, but: categorical can basically be everything (so object would still be generic, that is true), but `object` is actually not a proper callable like the other types (you cannot do `object(1)`, like you can do `int(1)`, `np.int64(1)`, or `Period(..)` ), so it doesn't follow the scheme.

In principle we could pass through the `type` of the categories, but the problem is that those can be None.

Since the type is actually never used (for the extension dtypes), I would also not bother too much about this one inconsistency.

Reply:
Joris is correct.

The only additional comment is that having a `.type` of object doesn't really tell you anything useful. But something like `Period`, `Interval`, or `Timestamp` actually is meaningful.

Reply:
so why don't we return None then? I think it is going to be very confusing that *only* Categorical as a DtypeType, but nothing else does. If you are getting rid of it, then remove it completely.

Reply:
None has the same drawbacks of object (cannot be called, is overly generic, does not hold any useful information). The CategoricalDtypeType at least has in its name it is the type of the CategoricalDtype

Reply:
Agreed with Joris.

Reply:
ok, then